cmd/manta-adm.js

```diff
@@ -967,6 +968,13 @@ MantaAdm.prototype.do_update.options = [
     'type':  'bool',
     'help': 'Allow deployment of experimental services'
 },
+{
+	'names': [ 'skip-verify-remote' ],
```

Comment:
Why the extra skip-verify-remote option? I don't see the utility beyond what 'skip-verify-channel' provides and it adds some complexity.

- if an operator can't reach updates.jo (air gapped or whatever), then they need to specific `--skip-verify-channel` already
- What would be the case for specifying `--skip-verify-remote` but NOT `--skip-verify-channel`?
- Couldn't an operator be surprised that specifying `--skip-verify-remote` implies that the channel will not be checked. Many will not make the connection that channel info isn't kept on a locally imported image.

Reply:
Right, I agree with some of that: skip-verify-remote confusingly does imply skip-verify-channel. The reverse isn't true.

Developers would still need `--skip-verify-channel` (eg. when upgrading to dev image on a system with the "release" channel set) Now imagine the images they want to provision also aren't available locally yet, so our "check for local images" error message would inform them of that.

In that scenario, I'd rather _also_ tell them that they can't deploy "mantav1-foo" here as early as possible, rather than simply not doing any remote lookups (if I understand you correctly, you were thinking that skip-verify-channel would also not do any remote lookups) Otherwise, developers will go to the trouble of downloading all of the images locally, only _then_, we'd tell them that they can't use the bits they downloaded, and we're back to whack-a-mole error states, which is a shame.

So what we could do is to require `--skip-verify-channel` if `--skip-verify-remote` is passed, that way, we're properly representing how the channel-checks work. Would that be better?

Reply:
> if I understand you correctly, you were thinking that skip-verify-channel would also not do any remote lookups

Nah (though apologies if I've changed my tune), I think we should just go ahead and look up the image remotely if we don't have it locally (sdcadm does this [here](https://github.com/joyent/sdcadm/blob/062fa02bf116990d6bb7ba18c8efbaa474fde2b4/lib/sdcadm.js#L1050-L1096)).   I.e. I agree with you that we should tell them about the image name mismatch as early as possible.

About the operator that has their Manta airgapped such that they cannot reach updates.joyent.com: let's cross that road if/when we come to it. I was being theoretical there. As well, I think for that case, this operator would only be specifying image UUIDs that they had already manually installed, so no real harm here.

Thoughts?

Reply:
Hmm, we can't just look up the image remotely only if the image isn't local, as sdcadm does - by default we always need to do a remote lookup to see if the image is in the correct channel, regardless of whether it exists locally as well.

Now, perhaps there's a case to be made for optimsation here: if we're told that we don't care about channel verification *and* all images exist locally, then we're wasting time doing remote lookups, but is that worth the extra complexity?

Reply:
The optimization would be nice, but I'm not going to stop this change for it. :)  
I don't think the code for it need feel complex. In stages:

- get all the new images locally
- if opts.skip_verify_channel, then filter down the set of remote images to get to just those we didn't find locally
- get the remote image manifests, if any left

Reply:
Yeah, it really wasn't too bad. I've added that, along with the CLI check that skipping remote should also imply skipping channel verification.

This looks like:
```
[root@0cf080f3-51de-4103-a13b-e55a8853083d (coal-1:manta0) ~]#  manta-adm update --skip-verify-remote new.json
manta-adm update: error: Channel verification requires remote imgapi lookups. Use --skip_verify_channel as well as --skip-verify-remote.
[root@0cf080f3-51de-4103-a13b-e55a8853083d (coal-1:manta0) ~]#
```

Then doing an update where we ought to reprovision two services, but only have one local image:

```
[root@0cf080f3-51de-4103-a13b-e55a8853083d (coal-1:manta0) ~]# manta-adm update new-one-local-one-remote-correct-channel.json
logs at /var/log/manta-adm.log
manta-adm: The following images are not available in the local imgapi instance:
    29bcf204-59c7-4292-a1e8-ef3e10d5ff52 (webapi)

with log entries:

[2019-11-28T11:35:25.756Z] DEBUG: manta-adm/6660 on 0cf080f3-51de-4103-a13b-e55a8853083d: getting local image 94ea1644-a329-4e45-9cb6-f5a5fc102f87 for service storage
[2019-11-28T11:35:25.757Z] DEBUG: manta-adm/6660 on 0cf080f3-51de-4103-a13b-e55a8853083d: getting local image 29bcf204-59c7-4292-a1e8-ef3e10d5ff52 for service webapi
[2019-11-28T11:35:25.807Z] DEBUG: manta-adm/6660 on 0cf080f3-51de-4103-a13b-e55a8853083d: missing local image 29bcf204-59c7-4292-a1e8-ef3e10d5ff52 (webapi)
[2019-11-28T11:35:25.808Z] DEBUG: manta-adm/6660 on 0cf080f3-51de-4103-a13b-e55a8853083d: getting remote image 94ea1644-a329-4e45-9cb6-f5a5fc102f87 for service storage on channel release
[2019-11-28T11:35:25.811Z] DEBUG: manta-adm/6660 on 0cf080f3-51de-4103-a13b-e55a8853083d: getting remote image 29bcf204-59c7-4292-a1e8-ef3e10d5ff52 for service webapi on channel release
```

and now the same check, this time when skipping channel verification, we only look for one remote image:

```
[root@0cf080f3-51de-4103-a13b-e55a8853083d (coal-1:manta0) ~]#  manta-adm update --skip-verify-channel new-one-local-one-remote-correct-channel.json
logs at /var/log/manta-adm.log
manta-adm: The following images are not available in the local imgapi instance:
    29bcf204-59c7-4292-a1e8-ef3e10d5ff52 (webapi)

and the log entries:

[2019-11-28T11:38:15.133Z] DEBUG: manta-adm/7275 on 0cf080f3-51de-4103-a13b-e55a8853083d: getting local image 94ea1644-a329-4e45-9cb6-f5a5fc102f87 for service storage
[2019-11-28T11:38:15.133Z] DEBUG: manta-adm/7275 on 0cf080f3-51de-4103-a13b-e55a8853083d: getting local image 29bcf204-59c7-4292-a1e8-ef3e10d5ff52 for service webapi
[2019-11-28T11:38:15.144Z] DEBUG: manta-adm/7275 on 0cf080f3-51de-4103-a13b-e55a8853083d: missing local image 29bcf204-59c7-4292-a1e8-ef3e10d5ff52 (webapi)
[2019-11-28T11:38:15.144Z] DEBUG: manta-adm/7275 on 0cf080f3-51de-4103-a13b-e55a8853083d: not searching remote imgapi for 94ea1644-a329-4e45-9cb6-f5a5fc102f87, it exists locally
[2019-11-28T11:38:15.145Z] DEBUG: manta-adm/7275 on 0cf080f3-51de-4103-a13b-e55a8853083d: getting remote image 29bcf204-59c7-4292-a1e8-ef3e10d5ff52 for service webapi on channel null
[2019-11-28T11:38:17.332Z] DEBUG: manta-adm/7275 on 0cf080f3-51de-4103-a13b-e55a8853083d: missing remote image 29bcf204-59c7-4292-a1e8-ef3e10d5ff52 (webapi)
```

Reply:
From discussion offline earlier, Trent felt that the option to skip the remote lookup altogether was too much complexity.

So now, we'll only do remote lookups if the local image isn't present, and in that case, we're already in an error scenario (no local image)

If we happen to also fail to do the remote lookup (eg. due to an air-gapped network) then there's still no chance of us deploying a badly-named uuid as a result of an image name we couldn't lookup, so the system is still correct.